heartbeat/sapdb.sh

```diff
@@ -44,11 +44,11 @@ background_check_saphostexec() {
 #                       misbehavior
 #
 cleanup_saphostexec() {
-  pkill -9 -f "$SAPHOSTEXEC"
-  pkill -9 -f "$SAPHOSTSRV"
-  oscolpid=`pgrep -f "$SAPHOSTOSCOL"`       # we check saposcol pid, because it
-                                            # might not run under control of
-					    # saphostexec
+  pkill -9 -f saphostexec
+  pkill -9 -f sapstartsrv
```

Comment:
Ah, I see.. I didn't realise that multiple instances could be active at once.

The process in particular that was the issue in this case was `saphostexec`. From looking at the resource agent, it looks to me like it expects the process to potentially be started by someone else. This was indeed the case that we encountered, only that the command line used to start it differed. As a result the SAPDatabase agent would stop the running instance and try to launch a new copy.

But, it is entirely possible that it is a question of misconfiguration / administrator error rather than something that really should be fixed.